app/helpers/votes_helper.rb

```diff
@@ -0,0 +1,19 @@
+module VotesHelper
+  def upvote_link(parent, user)
+    vote = parent.vote_by(user)
+    if vote && vote.upvote?
+      'Upvoted'
+    else
+      link_to 'Upvote', [parent, :upvote], method: :post
+    end
+  end
+
+  def downvote_link(parent, user)
+    vote = parent.vote_by(user)
+    if vote && vote.downvote?
+      'Downvoted'
+    else
+      link_to 'Downvote', [parent, :downvote], method: :post
+    end
+  end
+end
```

Comment:
Usually, we would want to write tests for these methods, too. Helper specs go in the helpers directory and are named after the module: 

`spec/helpers/votes_helper_spec.rb`

```ruby
require 'rails_helper'

RSpec.describe VotesHelper, type: :helper do
  describe "#upvote_link" do
    # tests go here
    # expect(upvote_link(article, user)).to eq('Upvoted')
  end
end
```